frontend/src/components/common/Calendar/WeeklyEventItemPopUp.tsx

```diff
@@ -0,0 +1,279 @@
+        donor.id,
+      );
+      setNextDropOff(
+        scheduleResponse.find((nextSchedule) => {
+          return (
+            new Date(nextSchedule!.startTime) > new Date(schedule!.startTime)
+          );
+        })?.startTime,
+      );
+    };
+
+    getNextDropOff();
+  }, []);
+
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent
+          maxW="90%"
+          width="fit-content"
+          maxH="fit-content"
+          height="fit-content"
+        >
+          <ModalHeader>
+            <Text
+              textStyle={isMobile ? "mobileHeader3" : "desktopSubtitle"}
+              pl={isMobile ? "1.125rem" : "6rem"}
+              pt={isMobile ? "2rem" : "4rem"}
+            >
+              Donation Details
+            </Text>
+          </ModalHeader>
+          <ModalCloseButton />
+          <Divider orientation="horizontal" />
+          <ModalBody>
+            <VStack
+              alignItems="start"
+              pl={isMobile ? "0.125rem" : "5rem"}
+              pt="2.5rem"
+            >
+              <Flex width="100%" pl="1rem">
+                <VStack alignItems="start">
+                  <Text textStyle={isMobile ? "mobileBodyBold" : "desktopBodyBold"} py="0.5rem">
+                    {donor.businessName
+                      ? donor.businessName
+                      : `${donor.firstName} ${donor.lastName}`}
+                  </Text>
+                  <HStack py="0.5rem">
+                    <CalendarIcon />
+                    <Text textStyle={isMobile ? "mobileSmall" : "desktopSmall"}>
+                      {new Date(schedule!.startTime).toLocaleString(undefined, {
+                        year: "numeric",
+                        month: "long",
+                        day: "numeric",
+                      })}
+                    </Text>
+                  </HStack>
+                  <HStack py="0.5rem">
+                    <TimeIcon />
+                    <Text textStyle={isMobile ? "mobileSmall" : "desktopSmall"}>
+                      {convertTime(schedule!.startTime)} -{" "}
+                      {convertTime(schedule!.endTime)}
+                    </Text>
+                  </HStack>
+                  {nextDropOff && (
+                    <Text textStyle={isMobile ? "mobileSmall" : "desktopSmall"} py="0.5rem">
+                      Next Dropoff:{" "}
+                      {new Date(nextDropOff).toLocaleString(undefined, {
+                        year: "numeric",
+                        month: "long",
+                        day: "numeric",
+                      })}
+                    </Text>
+                  )}
+                  {isMobile && (
+                    <Badge
+                      color={`${(colorMap as any)[schedule!.frequency]}.100`}
+                      backgroundColor={`${(colorMap as any)[schedule!.frequency]
+                        }.200`}

```

Comment:
I think the color map needs to be edited to be "One time donation" for the key

Reply:
Oh okay, lemme check with the designers to see what they finally decided on